vmm/src/signal_handler.rs

```diff
@@ -172,26 +178,33 @@ mod tests {
             // Call the blacklisted `SYS_mkdirat`.
             unsafe { syscall(libc::SYS_mkdirat, "/foo/bar\0") };
 
-            assert!(cpu_count() > 0);
-
-            // Kcov somehow messes with our handler getting the SIGSYS signal when a bad syscall
-            // is caught, so the following assertion no longer holds. Ideally, we'd have a surefire
-            // way of either preventing this behaviour, or detecting for certain whether this test is
-            // run by kcov or not. The best we could do so far is to look at the perceived number of
-            // available CPUs. Kcov seems to make a single CPU available to the process running the
-            // tests, so we use this as an heuristic to decide if we check the assertion.
-            if cpu_count() > 1 {
-                // The signal handler should let the program continue during unit tests.
-                assert_eq!(METRICS.seccomp.num_faults.count(), 1);
+            // Call SIGBUS signal handler.
```

Comment:
This still does not test the `sigbus_sigsegv_handler` function and I also find it relatively hard to properly test that function.

The purpose of the handler is to intercept the sigbus and sissegv signals so we can flush the metrics and logs, and then exit with a specific exit code. A good test function would test that these events happen. But since we cannot really call `libc::exit` because that would kill the `cargo test` process, we cannot really test the handlers. See: https://github.com/firecracker-microvm/firecracker/blob/18429c915f4786a23e615d24d1cc8213382b3a54/vmm/src/signal_handler.rs#L89

Reply:
If the metrics get incremented, it's proof enough that the signal handler was called. Like you said, the whole functionality can't be unit tested because of the `exit` thing.